servicetalk-http-netty/src/test/java/io/servicetalk/http/api/ServiceTalkContentCodingTest.java

```diff
@@ -0,0 +1,250 @@
+
+                        assertEquals(payloadAsString((byte) 'a'), requestPayload);
+
+                        final List<String> actualReqAcceptedEncodings = stream(request.headers()
+                                .get(ACCEPT_ENCODING, "NOT_PRESENT").toString().split(","))
+                                .map((String::trim)).collect(toList());
+
+                        final List<String> expectedReqAcceptedEncodings = clientSupportedEncodings.stream()
+                                .filter((enc) -> enc != identity())
+                                .map((ContentCodec::name))
+                                .map(CharSequence::toString)
+                                .collect(toList());
+
+                        if (reqEncoding != identity()) {
+                            assertTrue("Request encoding should be present in the request headers",
+                                    contentEquals(reqEncoding.name(),
+                                            request.headers().get(ACCEPT_ENCODING, "NOT_PRESENT")));
+                        }
+
+                        if (!expectedReqAcceptedEncodings.isEmpty() && !actualReqAcceptedEncodings.isEmpty()) {
+                            assertThat(actualReqAcceptedEncodings, equalTo(expectedReqAcceptedEncodings));
+                        }
+
+                        return super.handle(ctx, request, responseFactory);
+                    } catch (Throwable t) {
+                        t.printStackTrace();
+                        return succeeded(responseFactory.badRequest());
+                    }
+                }
+            };
+        }
+    };
+
+    static final Function<Scenario, StreamingHttpClientFilterFactory> RESP_VERIFIER = (scenario)
+            -> new StreamingHttpClientFilterFactory() {
+        @Override
+        public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
+            return new StreamingHttpClientFilter(client) {
+                @Override
+                protected Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
+                                                                final HttpExecutionStrategy strategy,
+                                                                final StreamingHttpRequest request) {
+                    return super.request(delegate, strategy, request).map(response -> {
+                        List<ContentCodec> server = scenario.serverSupported;
+                        List<ContentCodec> client = scenario.clientSupported;
+
+                        ContentCodec expected = identity();
+                        for (ContentCodec codec : client) {
+                            if (server.contains(codec)) {
+                                expected = codec;
+                                break;
+                            }
+                        }
+
+                        assertEquals(expected, encodingFor(client, response.headers()
```

Comment:
Similar logic here. If you verify in the filter, it can be executed in offloaded thread. You need to verify results on the caller side. Read the response via client API.